core/src/main/scala/kafka/coordinator/transaction/ProducerIdManager.scala

```diff
@@ -167,7 +167,7 @@ class RPCProducerIdManager(brokerId: Int,
       if (nextProducerId > currentProducerIdBlock.lastProducerId) {
         val block = nextProducerIdBlock.poll(maxWaitMs, TimeUnit.MILLISECONDS)
         if (block == null) {
-          throw Errors.REQUEST_TIMED_OUT.exception("Timed out waiting for next producer ID block")
+          throw Errors.CONCURRENT_TRANSACTIONS.exception("Timed out waiting for next producer ID block")
```

Comment:
Should we do this fix separately? I thought we had considered whether a short KIP would be needed. We probably also need to consider the behavior of other clients (such as rdkafka).

Reply:
Hmm -- I may have misunderstood some of our offline discussions. I believe a KIP is needed to bump the request version but since concurent transactions is already handled by the client for this request, my understanding is that it is fine.